torch/fft/__init__.py

```diff
@@ -9,18 +9,283 @@
+        normalization factor of ``1/n`` is applied. The mode determines where
+        this normalization is applied:
+
+        * ``"forward"`` - the forward transform is normalized by ``1/n``, backward is unnormalized.
+        * ``"backward"`` - the backward transform is normalized by ``1/n``, forward is unnormalized.
+        * ``"ortho"`` - both transforms are normalized by ``1/sqrt(n)`` (making it orthonormal).
+
+        Default is ``"backward"``.
+
+Example:
+
+    >>> import torch.fft
+    >>> t = torch.tensor([ 6.+0.j, -2.+2.j, -2.+0.j, -2.-2.j])
+    >>> torch.fft.ifft(t)
+    tensor([0.+0.j, 1.+0.j, 2.+0.j, 3.+0.j])
+""")
+
+rfft = _add_docstr(_fft.fft_rfft, r"""
+rfft(input, n=None, dim=-1, norm=None) -> Tensor
+
+Computes the one dimensional discrete Fourier transform of real :attr:`input`.
+
+The FFT of a real signal is Hermitian-symmetric, ``X[i] = conj(X[-i])`` so
+the output contains only the positive frequencies below the Nyquist frequency.
+To compute the full output, use :func:`~torch.fft.fft`
+
+Args:
+    input (Tensor): the real input tensor
+    n (int, optional): Signal length. If given, the input will either be zero-padded
+        or trimmed to this length before taking the real FFT.
+    dim (int, optional): The dimension along which to take the one dimensional real FFT.
+    norm (str, optional): Normalization mode. Between the forward and backward
+        transforms (:func:`~torch.fft.rfft` and :func:`~torch.fft.irfft`) a total
+        normalization factor of ``1/n`` is applied. The mode determines where
+        this normalization is applied:
+
+        * ``"forward"`` - the forward transform is normalized by ``1/n``, backward is unnormalized.
+        * ``"backward"`` - the backward transform is normalized by ``1/n``, forward is unnormalized.
+        * ``"ortho"`` - both transforms are normalized by ``1/sqrt(n)`` (making it orthonormal).
+
+        Default is ``"backward"``.
+
+Example:
+
+    >>> import torch.fft
+    >>> t = torch.arange(4)
+    >>> t
+    tensor([0, 1, 2, 3])
+    >>> torch.fft.rfft(t)
+    tensor([ 6.+0.j, -2.+2.j, -2.+0.j])
+
+    Compare against the full output from :func:`~torch.fft.fft`:
+
+    >>> torch.fft.fft(t)
+    tensor([ 6.+0.j, -2.+2.j, -2.+0.j, -2.-2.j])
+
+    Notice that the symmetric element ``T[-1] == T[1].conj()`` is omitted.
+    At the Nyquist frequency ``T[-2] == T[2].conj() == T[2]`` is given but must
+    always be purely real.
+""")
+
+irfft = _add_docstr(_fft.fft_irfft, r"""
+irfft(input, n=None, dim=-1, norm=None) -> Tensor
+
+Computes the inverse of :func:`~torch.fft.rfft`.
+
+:attr:`input` is a one-sided Hermitian signal in the Fourier domain, as
+produced by :func:`~torch.fft.rfft`. By the Hermitian property, the time-domain
+signal will be purely real.
+
+Note:
+    Some input frequencies must be purely real to satisfy the Hermitian
+    property. In these cases the imaginary component will be ignored.
```

Comment:
I'm a little confused how this sentence relates to the previous one. 

This function is interpreting input as a one-sided Hermitian signal in the Fourier domain. Is this saying that if that input doesn't satisfy an expected invariant then it will be interpreted as if it did?

Reply:
Yes, for example if the the zero-frequency term had an imaginary component then it would be impossible to include that imaginary term in the real output. So it would always just be ignored.

Reply:
Got it. What about adding a brief, "For example, ..." sentence after at the end of this to clarify just like you did here?

Reply:
Added the example to this and `hfft`.